### PR TITLE
Farza/feature/20161107/improve gulp tasks

### PR DESCRIPTION
**Pull request by arzafran**
Added gulp doc:watch task to start documenting like champs! :+1:

### CODE REVIEW

**Comment by damianmuti** [comment]
Works great, buddy! 💯

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = false
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (andresmoschini) [inline .editorconfig]: Why?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
```
Comment (andresmoschini) [inline .editorconfig]: I ❤️ EditorConfig

**Comment by andresmoschini** [comment]
Nice!

File: .editorconfig (hunk 1 above)
Comment (arzafran) [inline .editorconfig]: This is the convention for the MSUIF files :)

Comment (andresmoschini) [inline .editorconfig]: I understand, but, by curiosity, why would you want to keep trailing white spaces in markdown files?

File: gulpfile.js [hunk 3]
```diff
@@ -167,103 +184,78 @@ gulp.task('webfont:generate', function(){
...
-      port: 3000,
-      user: '',
-      pass: '',
-      remotePath: ''
-    }));
+gulp.task('doc', function() {
+  var docstream = sassdoc(config.sassDocOptions);
+  gulp.src(config.folderAssets.base + '/**/*.scss')
+    .pipe(docstream);
+  return docstream.promise.then(browserSync.reload);
```
Comment (andresmoschini) [inline gulpfile.js]: I am not familiar with gulp, but, comparing with the previous code, I have a doubt.

Is it fine to return a promise here? or should you return the result of `gulp.src(...).pipe(...)`?

File: .editorconfig (hunk 1 above)
Comment (arzafran) [inline .editorconfig]: You're right, copy and paste has failed me! haha

**Comment by joseguti84** [comment]
💯

**Pull request merged**